agent/provider/src/payments/pricing.rs

```diff
@@ -52,7 +52,8 @@ impl PaymentModel for LinearPricing {
                 .zip(usage.iter())
                 .map(|(coeff, usage_value)| coeff * usage_value)
                 .sum::<f64>();
-        Ok(BigDecimal::from(cost))
+
+        BigDecimal::from_f64(cost).ok_or(anyhow!("Failed to convert to BigDecimal: {}", cost))
```

Comment:
I don't understand this change. Why use `from_f64` if you can use `from` and skip the unnecessary error handling? This operation should always succeed.

Reply:
Ah, I see, they removed it.